src/main/java/com/algorand/algosdk/transaction/Transaction.java

```diff
@@ -901,6 +903,14 @@ public boolean equals(Object o) {
         @JsonProperty("t")
         public BigInteger assetTotal = BigInteger.valueOf(0);
 
+        // Decimals specifies the number of digits to display after the decimal
+        // place when displaying this asset. A value of 0 represents an asset
+        // that is not divisible, a value of 1 represents an asset divisible
+        // into tenths, and so on. This value must be between 0 and 19
+        // (inclusive).
+        @JsonProperty("dc")
+        public BigInteger assetDecimals = BigInteger.valueOf(0);
```

Comment:
assetDecimals type better be byte, short or int, and not BigInteger. It holds only 19 values, and it is implemented as 32 bit int on the server side. 
Primitive types are easier to handle in the code than BigInteger. Let us not add more of it. At some point, it should all go away.